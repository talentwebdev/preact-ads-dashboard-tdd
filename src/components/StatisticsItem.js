import React from "react";
import BarChartIcon from "@material-ui/icons/BarChart";
function StatisticsItem({ title, subTitle1, subTitle2, value1, value2 }) {
  return (
    <div className="viewItem">
      <div className="itemTitle">
        <p>{title}</p>
      </div>
      <div className="itemContent">
        <div className="itemContentLeft">
          <p>{subTitle1}</p>
          <h3>{value1}</h3>
        </div>
        <div className="itemContentRight">
          <p>{subTitle2}</p>
          <h3>{value2}</h3>
        </div>
      </div>
      <div className="itemFooter">
        <div className="itemFooterTitle">
          <p>View Statistics</p>
        </div>
        <div className="itemFooterIcon">
          <BarChartIcon />
        </div>
      </div>
    </div>
  );
}

export default StatisticsItem;
