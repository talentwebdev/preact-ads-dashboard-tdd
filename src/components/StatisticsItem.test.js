import { shallow } from "enzyme";
import StatisticsItem from "./StatisticsItem";

describe("StatisticsItem", () => {
  let statisticsItemWrapper;
  beforeAll(() => {
    statisticsItemWrapper = shallow(<StatisticsItem />);
  });

  it("renders elements", () => {
    expect(statisticsItemWrapper.find("div")).toHaveLength(8);
  });
  it("renders StatisticsItem correctly", () => {
    expect(statisticsItemWrapper).toMatchSnapshot();
  });
});
