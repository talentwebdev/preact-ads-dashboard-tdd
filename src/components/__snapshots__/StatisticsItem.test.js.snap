// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsItem renders StatisticsItem correctly 1`] = `ShallowWrapper {}`;
